### ISSUE TITLE
Puzzle list page scroll jumps to top on resize

### ISSUE THREAD

**Issue by grantaelliott** [deathandmayhem/jolly-roger#376]
This issue appears to affect Safari only. On the puzzle list page, if the search box has focus but is out of the viewport and the width of the viewport changes, the scroll will jump to the top of the page.

**Comment by zarvox** [comment]
Curious.  I'm unable to reproduce this on our production instance on desktop Safari version 14.0.2 (16610.3.7.1.9).

**Comment by zarvox** [comment]
Grant, were you ever able to reproduce this?

**Comment by grantaelliott** [comment]
Yes. This issue persists viewing the production site in Safari for Mac.

**Comment by zarvox** [comment]
I remain unable to reproduce this on an M1 MacBook Air at the same version mentioned above ( 14.0.2 (16610.3.7.1.9) ):

https://user-images.githubusercontent.com/307325/104255821-c8431380-542e-11eb-93de-7d0bcdfe476d.mov

**Comment by grantaelliott** [comment]
I've verified that other users (on Intel Macs) can also reproduce this.

@zarvox Does the "Filter by title, answer, or tag" input have focus? It seems to only happen when that is true.

**Comment by zarvox** [comment]
Oh -- yes.  I meant to include the blinking cursor and blue outline in the screen recording, but must have goofed it up somewhere.  But yes, that field is focused.

**Comment by zarvox** [comment]
@grantaelliott Are you still able to reproduce this issue?  I would love for this to not trip up Safari users, but I'm still unable to reproduce this as described.

**Comment by grantaelliott** [comment]
I can still reproduce this in production on Safari 16.2

**Comment by zarvox** [comment]
Hmm.  I just did the update to macOS 13.1 with Safari 16.2 (18614.3.7.1.5) on my M1 MacBook Air, and I am still unable to reproduce the issue.  When the filter bar has keyboard focus, I am able to scroll the page freely with both the trackpad and an external mouse, and then when I resize the window with either, the page seems to preserve the same scroll offset.  If I press a key (changing the filter), only then does the page scroll the filter input back into view.

Do you have any Safari extensions enabled, or some sort of accessibility tool or setting that could be causing simulated input?  I note that when I type into the input the input bar scrolls into view, but I'm having a hard time imagining what else would be triggering this.

Unfortunately, I lack access to an Intel Mac at present, so I'm going to have a hard time debugging this myself.  As a long shot, to see if there's something about your local configuration that contributes to this: if you create a new macOS user and then log in as that user, does it still happen?

**Comment by grantaelliott** [comment]
I just verified on another Intel Mac. I’ve had a number of people try it in the past as well. In fact, I just discovered this also occurs in Safari on iPadOS.

I agree that this feels like it could be an accessibility option or just an Apple-specific focus behavior, but it scrolls entirely to the top of the page, not just to the input. I also confirmed that it doesn’t happen on other sites. Whatever the cause, it’s very strange that it doesn’t seem to affect Apple silicon Macs.

It’s a very low priority issue, but sometime I will try to diagnose it.

**Comment by grantaelliott** [comment]
I've found that when an offscreen input field with focus changes width, Safari attempts to vertically center it in the viewport by scrolling. In our case, the input's width changes when the viewport is resized horizontally and attempting to center it just scrolls to the page start for sensible viewport heights.

MRE - Focus on the input, scroll down, and resize the viewport horizontally:
```
<html>
  <body>
    <input style="width:100%;" />
    <div style="height:2000px;"></div>
  </body>
</html>
```

Why Safari does this (and why it apparently doesn't do it on Apple Silicon Macs), I don't know.

This affects the puzzle list and hunters list pages. Unfortunately, I don't think there's much to be done about it on a responsive page.  Closing as won't fix.
